business/istio_validations.go

```diff
@@ -258,41 +185,79 @@ func runObjectCheckers(objectCheckers []ObjectChecker) models.IstioValidations {
 	return objectTypeValidations
 }
 
+// The following idea is used underneath: if errChan has at least one record, we'll effectively cancel the request (if scheduled in such order). On the other hand, if we can't
+// write to the buffered errChan, we just ignore the error. This is because a single error is enough to cancel the whole request.
+
```

Comment:
Goland is marking that next "func fetch()" is not used anymore and the following "func fetchNoEntry()" is only used in a single point for "getGatewaysPerNamespace". So I suggest to remove the unused code, and perhaps to not use the fetchNoEntry and call it directly in the gateways context.

Reply:
The same function is used in the istioDetails fetching (in kubernetes) package, so I'd rather not remove / rename the fetchNoEntry at this point - also it makes the fetchGateways too large in my opinion if it's embedded there. I don't have the fetch error for some reason in my IDE, but it is indeed unused here.